pkg/apis/security/v1alpha1/common_validation.go

```diff
@@ -112,9 +118,19 @@ func (pbs *PolicyBindingSpec) Validate(ctx context.Context, parentNamespace stri
 	if pbs.Subject.Namespace != parentNamespace {
 		errs = errs.Also(apis.ErrInvalidValue(pbs.Subject.Namespace, "namespace").ViaField("subject"))
 	}
+	// Specify subject either by an object reference or by label selector.
+	// Cannot be both.
+	if pbs.Subject.APIVersion != "" || pbs.Subject.Kind != "" || pbs.Subject.Name != "" {
+		if pbs.Subject.APIVersion == "" || pbs.Subject.Kind == "" || pbs.Subject.Name == "" {
```

Comment:
this confuses me a bit, don't find it very readable. Anyway we can revisit this logic? 
Maybe some inspiration: https://github.com/knative/eventing/blob/8ce1b8084c9055ae24c7b0d8ca22f99d6907558d/pkg/apis/messaging/v1beta1/subscribable_channelable_validation.go#L37